heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraScheduler.java

```diff
@@ -184,9 +185,29 @@ public boolean onUpdate(Scheduler.UpdateTopologyRequest request) {
     return true;
   }
 
+
+  boolean confirmWithUser(int newContainerCount) {
+    LOG.info(String.format("After update there will be %d more containers. "
+        + "Please make sure there are sufficient resources to update this job. "
+        + "Continue update? [y/N]: ", newContainerCount));
+    Scanner scanner = new Scanner(System.in);
+    String userInput = scanner.nextLine();
+    if ("y".equalsIgnoreCase(userInput)) {
+      return true;
+    }
+    return false;
+  }
+
   @Override
   public Set<PackingPlan.ContainerPlan> addContainers(
       Set<PackingPlan.ContainerPlan> containersToAdd) {
+    if ("prompt".equalsIgnoreCase(Context.updatePrompt(config))) {
+      if (!confirmWithUser(containersToAdd.size())) {
+        LOG.warning("Scheduler updated topology canceled.");
+        return null;
```

Comment:
return null sometimes will cause an additional null check logic when using this method.
In the effective java design pattern, it should always avoid returning null.
Or you can return an empty object or initial the object at the very beginning.
ag, I don't know the whole logic, just some notes.

Reply:
updated